Helper/InsightsHelper.php

```diff
@@ -157,6 +157,30 @@ public function setAuthenticatedUserToken(Customer $customer): string|null
         return $userToken;
     }
 
+    /**
+     * Sets a token for a non-authenticated user
+     *
+     * @return void
+     */
+    public function setNonAuthenticatedUserToken(): void
+    {
+        $metaData = $this->cookieMetadataFactory->createPublicCookieMetadata()
+            ->setDuration($this->configHelper->getCookieLifetime())
```

Comment:
I'm not sure we want to refresh this token's expiration at this time. By default it is [six months](https://www.algolia.com/doc/guides/sending-events/concepts/usertoken/#adjust-the-cookie-expiration-time) and having a long lived anonymous cookie has value for tracking perso data for those who take a long time before making a purchase decision.